README.md

```diff
@@ -394,10 +396,57 @@ The `extract` method takes no arguments:
     1. `v` must have the same type that `f` returns in `extend`.
 
 
+### Bifunctor
 
+A value that implements the Bifunctor specification is a functor with two
+type arguments that are both covariant.
 
+1. `p.bimap(a => a, b => b)` is equivalent to `p` (identity)
+2. `p.bimap(compose(f1)(f2), compose(g1)(g2))` is equivalent to `p.bimap(f1, g1).bimap(f2, g2)` (composition)
 
+#### `bimap` method
 
+A value which has a Bifunctor must provide an `bimap` method. The `bimap`
+method takes two arguments:
+
+    c.bimap(f, g)
+
+1. `f` must be a function which returns a value
+
+    1. If `f` is not a function, the behaviour of `bimap` is unspecified.
+
+2. `g` must be a function which returns a value
+
+    1. If `g` is not a function, the behaviour of `bimap` is unspecified.
+
+
+### Profunctor
+
+A value that implements the Profunctor specification must also implement
+the Functor specification. The Profunctor specification is a contravariant
+functor on its first type parameter and covariant functor on its second type
+parameter.
+
+1. `p.promap(a => a, b => b)` is equivalent to `p` (identity)
+2. `p.promap(compose(f1)(f2), compose(g1)(g2))` is equivalent to `p.promap(f1, g1).promap(f2, g2)` (composition)
```

Comment:
The order of composition seems wrong. Should be:

```js
p.promap(compose(f1)(f2), compose(g2)(g1)) ≡ p.promap(f1, g1).promap(f2, g2)
```

Also we probably should use "inlined" function composition like in the rest of the spec, and I would use four letters instead of letters with numbers, so:

```js
p.promap(x => f(g(x)), x => h(i(x))) ≡ p.promap(f, i).promap(g, h)
```

I took letters from https://hackage.haskell.org/package/profunctors-5.2/docs/Data-Profunctor.html